static/app/stories/view/storySearch.tsx

```diff
@@ -1,103 +1,201 @@
-            };
-          }}
-          placeholder="Search stories"
-          defaultValue={location.query.query ?? ''}
-          onChange={onSearchInputChange}
-        />
-        <InputGroup.TrailingItems>
-          <Badge type="internal">/</Badge>
-        </InputGroup.TrailingItems>
-        {/* @TODO (JonasBadalic): Implement clear button when there is an active query */}
-      </InputGroup>
-      {showSearch && (
-        <Overlay
-          style={{position: 'absolute', top: 0, left: 0}}
-          onClick={dismiss}
-          onKeyDown={e => {
-            if (e.key === 'Enter') {
-              dismiss();
-            }
-          }}
-        >
-          <StorySearchContainer>
-            <StoryTree nodes={searchTree} />
-          </StorySearchContainer>
-        </Overlay>
+    <SearchComboBox
+      label={t('Search stories')}
+      defaultItems={tree.map(node => {
+        return {id: node.filesystemPath, name: normalizeFilename(node.name)};
+      })}
+      defaultSelectedKey={defaultSelectedKey}
+      menuTrigger="focus"
+      inputRef={searchInput}
+    >
+      {item => (
+        <SearchItem key={item.id} textValue={item.name}>
+          <Text slot="label">{item.name}</Text>
+        </SearchItem>
       )}
-    </Fragment>
+    </SearchComboBox>
+  );
+}
+
+function SearchInput(props: InputProps & React.RefAttributes<HTMLInputElement>) {
+  [props, props.ref] = useContextProps(props, props.ref ?? {current: null}, InputContext);
+  const {className: _0, style: _1, size: nativeSize, ...nativeProps} = props;
+
+  return (
+    <InputGroup style={{minHeight: 33, height: 33, width: 256}}>
+      <InputGroup.LeadingItems disablePointerEvents>
+        <IconSearch />
+      </InputGroup.LeadingItems>
+      <InputGroup.Input ref={props.ref} nativeSize={nativeSize} {...nativeProps} />
+      <InputGroup.TrailingItems>
+        <Badge type="internal">/</Badge>
+      </InputGroup.TrailingItems>
+    </InputGroup>
+  );
+}
+
+interface SearchComboBoxProps<T extends Record<PropertyKey, unknown>>
+  extends Omit<ComboBoxProps<T>, 'children'> {
+  children: React.ReactNode | ((item: T) => React.ReactNode);
+  description?: string | null;
+  errorMessage?: string | ((validation: ValidationResult) => string);
+  inputRef?: React.RefObject<HTMLInputElement | null>;
+  label?: string;
+}
+
+function SearchComboBox<T extends Record<PropertyKey, unknown>>({
+  label,
+  errorMessage,
+  children,
+  inputRef,
+  ...props
+}: SearchComboBoxProps<T>) {
+  const navigate = useNavigate();
+  const handleSelectionChange = useCallback(
+    (key: string | number | boolean | null) => {
+      if (!key) {
+        return;
+      }
+      navigate(`/stories?name=${encodeURIComponent(key)}`, {replace: true});
+    },
+    [navigate]
+  );
+
+  return (
+    <ComboBox {...props} onSelectionChange={handleSelectionChange}>
+      <Label className="sr-only">{label}</Label>
+      <div className="my-combobox-container">
```

Comment:
`my-combobox-container`

Reply:
Removed in [969e419](https://github.com/getsentry/sentry/pull/94989/commits/969e419e9f13be1943b7bb841e76b3604c2a2693)